### DISCUSSION TITLE
I have absolutely no Idea what I am doing.

### DISCUSSION THREAD

**Discussion by AndroidGuy2** [bubuntux/nordvpn#180] [Q&A]
I am running this on a SynologyNAS DS918+.  My goal is to have Sonarr, Radarr, and SabNZBD routed through this bubuntux/nordvpn container.  But I have no idea how to get this set up.  Although I am familiar with putty, I have never set up a docker container using it.   So far on my 918+, I have successfully set up two containers.  
One of them is oznu-homebridge, which I set up using this video: https://www.youtube.com/watch?v=HF6XXTwJP0Y

and the other is openspeedtest, which I set up using this video: https://www.youtube.com/watch?v=8QkE0Avcehg

Neither scenario involved using ssh.  

So with that said, is there a guide somewhere for me to set it all up within my synology, and get my other three containers' traffic routed through it?  (and to clarify, I only want those containers routed through it, but as for the 918's traffic itself, I want to just flow uninhibited without going through VPN).

**Comment by guisua** [comment]
I wrote [this step by step guide](https://github.com/guisua/jdownloader-vpn-docker-synology) a few months back, which shows how to do exactly what you have in mind only with a jdownloader docker container as the "consumer" of the connection.

My guess:
- if you mention sonarr and radarr, you probably wouldn’t mind a jdownloader container either ;)
- I'm sure you can take the same lógico and apply it to create the other containers instead of jdownloader

Good luck and feel free to ask for more detailed help if you have any problems!

**Comment by AndroidGuy2** [comment]
So I am trying to follow your guide.  The container seems to be created okay, but I get this at the very end of it:

docker: Error response from daemon: linux runtime spec devices: error gathering device information while adding custom device "/dev/net/tun": no such file or directory.

How do I fix this?

**Comment by AndroidGuy2** [comment]
Okay, I used instructions from user EVOTk [here](https://www.synoforum.com/threads/device-dev-net-tun-not-working-anymore-after-docker-update-18-09-0-0513.3074/) to get the /dev/net/tun created.  
Now the creation of the container seems to go just fine.    But now, when I power the nordvpn module on, it doesn't stay on.  It stays on for maybe five seconds and then turns off.  I get a message in Synology DSM notifications that says "Docker container nordvpn stopped unexpectedly"

**Comment by AndroidGuy2** [comment]
I figured it out.  I was using my email address as my nord login.   But really what i needed to be using was the login username that nord gives you to set up stuff like openvpn.  
Thank you for your help.

When it's all said and done, how can i verify that traffic is indeed being routed through nord?

**Comment by AndroidGuy2** [comment]
On 2nd thought, I don't think that works.  So I switched my username back to my email.   But in the log i get
 [31mWhoops! We couldn't connect you to 'United_States'. Please try again. If the problem persists, contact our customer support.

**Comment by AndroidGuy2** [comment]
I finally read in this post that you need to use an older version 3.7.4
https://github.com/bubuntux/nordvpn/issues/103

Now I just need to figure out how to route other containers' traffic through this container.
I am using openspeedtest as a test subject, because I had already had it set up, and secondly because I want to test the speed of this Nord connection.   So I go in to edit the Environment of the openspeedtest and I add a variable "NETWORK" with a value of "container:nordvpn" ...  but this still doesn't seem to be working.

For the record, I can see in the log for the Nord container that it is connected okay and it got assigned a new IP from Nord.

**Comment by guisua** [comment]
@AndroidGuy2 I think (haven't tested it) setting the network only works when creating the container and not afterwards if the container already exists. Otherwise, I can confirm that version 3.7.4 is currently the latest working fine.

**Comment by guisua** [comment]
> 
> 
> Okay, I used instructions from user EVOTk [here](https://www.synoforum.com/threads/device-dev-net-tun-not-working-anymore-after-docker-update-18-09-0-0513.3074/) to get the /dev/net/tun created.
> Now the creation of the container seems to go just fine. But now, when I power the nordvpn module on, it doesn't stay on. It stays on for maybe five seconds and then turns off. I get a message in Synology DSM notifications that says "Docker container nordvpn stopped unexpectedly"

Just as a follow-up, didn't you see my mention of this in the [troubleshooting section](https://github.com/guisua/jdownloader-vpn-docker-synology#troubleshooting) or did what I have there not work (in that case I should update it). Thanks!

**Comment by CallMeTerdFerguson** [comment]
> I finally read in this post that you need to use an older version 3.7.4
> #103
> 
> Now I just need to figure out how to route other containers' traffic through this container.
> I am using openspeedtest as a test subject, because I had already had it set up, and secondly because I want to test the speed of this Nord connection. So I go in to edit the Environment of the openspeedtest and I add a variable "NETWORK" with a value of "container:nordvpn" ... but this still doesn't seem to be working.
> 
> For the record, I can see in the log for the Nord container that it is connected okay and it got assigned a new IP from Nord.

What you want is to use the NordVPN container's networking as the network for the other containers. The best way to do this, to ensure the failsafe works and the container doesn't fall back to another, non-vpn network, is not done via environment variables but via actual docker capabilities. I use docker-compose for all my needs, so I don't know the exact name of the equivalent docker flags you need are for manual container creation but what you need is to set the reliant containers `network_mode` to `service:<serviceName>` where serviceName is the name of the NordVPN container.  If you opt for compose, you'll also want to set the reliant containers to depend on the VPN container, just to ensure that docker knows that the VPN container needs to be running first. In compose it looks like this. You can look up in the docker doc what the equivalent command line options are if you manually create all of your containers.
```yml
     vpn:
       image: bubuntux/nordvpn:3.7.4 
       container_name: "nord-vpn"
       <<other config>>
     service-using-vpn:
       image: someimage/iwanttovpn
       depends_on:
         - vpn
       network_mode: "service:vpn"  
       <<other config>>
`
